proxy/m2_actions.c

```diff
@@ -1185,6 +1185,80 @@ _m2_container_create (struct req_args_s *args, struct json_object *jbody)
 	return _reply_m2_error (args, err);
 }
 
+static enum http_rc_e
+_m2_container_create_many (struct req_args_s *args, struct json_object *jbody)
+{
+	guint max_request = OIO_CONTAINER_CREATE_MANY_NB_MAX_REQ;
+	guint nb_err = 0;
+	GString *gresponse = g_string_sized_new(2048);
+	g_string_append(gresponse, "{\"containers\":[");
+	GError *err = NULL;
+	const char *url_user = oio_url_get(args->url, OIOURL_USER);
+	json_object *jarray = NULL;
+	if (!oio_url_get(args->url, OIOURL_ACCOUNT)
+		|| !oio_url_get(args->url, OIOURL_NS))
+		return _reply_format_error(args,
+				BADREQ("Missing account or namespace"));
+	if (!json_object_object_get_ex(jbody, "containers", &jarray)
+		|| !json_object_is_type(jarray, json_type_array))
+		return _reply_format_error(args,
+				BADREQ("Invalid array of containers"));
+	unsigned jarray_len = json_object_array_length(jarray);
+	if (jarray_len > max_request)
+		return _reply_format_error(args,
+			BADREQ("More than %u requested",
+			 max_request));
+	for (unsigned i= 0; i < jarray_len && !err; i++) {
+		struct json_object * jcontainer=NULL;
+		struct json_object * jname = NULL;
+		gchar **properties = NULL;
+		const gchar *name = NULL;
+		jcontainer = json_object_array_get_idx (jarray, i);
+		g_string_append (gresponse, "{\"name\":");
+		if(!json_object_object_get_ex(jcontainer, "name", &jname)){
+			return _reply_format_error(args,
+				BADREQ("No \"name\" field"));
+		}
+		if(!json_object_is_type(jname, json_type_string))
+			return _reply_format_error(args,
+				BADREQ("Bad \"name\" field"));
```

Comment:
You cannot break the loop without cleaning things, unless you want memory leaks... I advise you to do as you did below: save the error in the response and continue.